<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><title>Hui Jiang</title>

<link href="cgb.css" rel="stylesheet" type="text/css"></head>
<style type="text/css">
    .auto-style1 {
        text-decoration: underline;
    }
    .auto-style2 {
        font-size: small;
        font-style: italic;
    }
    .auto-style3 {
        font-size: small;
    }
    .auto-style4 {
        font-size: small;
        font-weight: bold;
    }
    .auto-style5 {
        width: 800px;
    }
</style>
<body>
<table border="0" width="436">
<tbody>
<tr>
<td width="141"><font size="5"><img src="portrait.JPG" height="207" width="140"></font></td>
<td width="242">
<p><font size="5"><font size="7">Hui
Jiang</font></font></p>
<p>Professor<br>
<a href="http://www.sph.umich.edu/biostat/">Department
of Biostatistics</a><br>
<a href="http://www.sph.umich.edu/">School of Public
Health</a><br>
<a href="http://www.umich.edu/">University of
Michigan</a></p>
</td>
</tr>
</tbody>
</table>
<hr>
<table border="0" width="960">
<tbody>
<tr>
<td width="590">
<p><strong>Contact Information</strong></p>
<p> M4523 SPH II<br>
1415 Washington Heights <br>
Ann Arbor, Michigan 48109-2029</p>
<p>Office: (734) 764-6742</p>
<p>E-mail: jianghui AT umich.edu </p>
<p><strong>Professional Summary</strong></p>
<p> Hui Jiang is a Professor in the Department
of Biostatistics at University of Michigan. He received his Ph.D. in Computational and
Mathematical Engineering from Stanford University. Before joining the University of Michigan,
he was a postdoctoral scholar in the Department of Statistics
and Genome Technology Center at Stanford University. He is interested in
developing statistical and computational methods for the analysis of
large-scale biological data generated using modern high-throughput
technologies.</p>
<p><strong>Research Interests</strong></p>
<ul>
<li><strong>Statistical Genomics and Bioinformatics</strong>
<ul>
<li> High-throughput genomics, sequencing data, RNA-Seq, gene expression, cancer genomics</li>
</ul>
</li>
<li> <strong>Computational Statistics</strong>
<ul>
<li>Statistical computing, optimization, resampling, high-dimensional data, penalized models, variable selection, machine learning</li>
</ul>
</li>
</ul>
<p><strong><a href="software.html">Software & Programs</a></strong></p>
<p><strong>Selected Publications <small>(<a href="http://scholar.google.com/citations?user=LszKw2cAAAAJ&amp;hl=en">Google Scholar</a>) </small>
</strong></p>
<ul>
  <li>
       <p class="p2">
           <span class="auto-style3">Tran L, Li G, Luo L, Jiang H (2024) A fast solution to the lasso problem with equality constraints</span><br class="auto-style3"><span class="auto-style2">Journal of Computational and Graphical Statistics</span><span class="auto-style3">, 33 (3): 804-813. [<a href="https://doi.org/10.1080/10618600.2023.2277877">online</a>]</span><span class="auto-style3">[<a href="https://github.com/lamttran/ECLasso">code</a>]</span></p>
   </li>
     <li>
      <p class="p2">
          <span class="auto-style3">Shi Y, Shi W, Wang M, Lee J, Kang H, Jiang H (2023) Accurate and fast small p-value estimation for permutation tests in high-throughput genomic data analysis with the cross-entropy method</span><br class="auto-style3"><span class="auto-style2">Statistical Applications in Genetics and Molecular Biology</span><span class="auto-style3">, 22 (1). </span><span class="s3">[<a href="https://www.degruyter.com/document/doi/10.1515/sagmb-2021-0067/html"><span class="auto-style3">online</span></a>][<a href="https://arxiv.org/abs/1608.00053"><span class="auto-style3">arXiv</span></a><span class="auto-style3">][<a href="https://github.com/shilab2017/Permutation-CE-codes"><span class="auto-style3">code</span></a>]</span></span></p>
  </li>
  <li>
       <p class="p2">
           <span class="auto-style3">Zhai J, Jiang H (2023) Two-sample test with g-modeling and its applications</span><br class="auto-style3"><span class="auto-style2">Statistics in Medicine</span><span class="auto-style3">, 42 (1): 89-104. [<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/sim.9603">online</a>]</span><span class="auto-style3">[<a href="https://github.com/kkttzjy/gmodeltest">code</a>]</span></p>
   </li>
   <li>
        <p class="p2">
            <span class="auto-style3">Tran L, He K, Wang D, Jiang H (2023) A Cross-Validation Statistical Framework for Asymmetric Data Integration</span><br class="auto-style3"><span class="auto-style2">Biometrics</span><span class="auto-style3">, 79 (2): 1280-1292. [<a href="https://doi.org/10.1111/biom.13685">online</a>]</span><span class="auto-style3">[<a href="https://github.com/lamttran/intasymm">code</a>]</span></p>
    </li>
    <li>
       <p class="p2">
           <span class="auto-style3">Zhai J, Ji H, Jiang H (2023) Reconstruction of Single-Cell Trajectories Using Stochastic Tree Search</span><br class="auto-style3"><span class="auto-style2">Genes</span><span class="auto-style3">, 14(2), 318. [<a href="https://www.mdpi.com/2073-4425/14/2/318">online</a>]</span><span class="auto-style3">[<a href="https://github.com/kkttzjy/STS">code</a>]</span></p>
   </li>
   <li>
       <p class="p2">
           <span class="s2"><span class="auto-style3">Zhu Y, Shen X, Jiang H, Wong WH (2023) Collaborative multilabel classification</span><br class="auto-style3"><span class="auto-style2">Journal of the American Statistical Association</span><span class="auto-style3">, 118 (542): 913-924. </span> </span><span class="s3">[<a href="https://doi.org/10.1080/01621459.2021.1961783"><span class="auto-style3">online</span></a><span class="auto-style3">]</span></span></p>
   </li>
  <li>
      <p class="p2">
          <span class="auto-style3">Shi Y, Lee J, Kang H, Jiang H (2022) A Two-part Mixed Model for Differential Expression Analysis in Single-cell High-throughput Gene Expression Data</span><br class="auto-style3"><span class="auto-style2">Genes</span><span class="auto-style3">, 13(2), 377. </span><span class="s3">[<a href="https://www.mdpi.com/2073-4425/13/2/377"><span class="auto-style3">online</span></a>]</span><span class="auto-style3">[<a href="https://github.com/shilab2017/two_part_mixed_model">code</a>]</span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Wu W, Taylor, JMG, Brouwer AF, Luo L, Kang J, Jiang H, He K (2022) Scalable Proximal Methods for Cause-Specific Hazard Modeling with Time-Varying Coefficients</span><br class="auto-style3"><span class="auto-style2">Lifetime Data Analysis</span><span class="auto-style3">, 28: 194-218. </span> </span><span class="s3">[<a href="https://link.springer.com/article/10.1007/s10985-021-09544-2"><span class="auto-style3">online</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
          <p class="p2">
              <span class="auto-style3">Liu K, Shen L, Jiang H (2019) Joint Between-Sample Normalization and Differential Expression Detection through L0-regularized Regression </span><br class="auto-style3"><span class="auto-style2">BMC Bioinformatics</span><span class="auto-style3">, 20(Suppl 16): 593. </span><span class="s3">[<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3070-4"><span class="auto-style3">online</span></a>]<span class="auto-style3">[<a href="rseqrobust/">code</a>]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="auto-style3">Shi Y, Wang M, Shi W, Lee J, Kang H, Jiang H (2019) Accurate and Efficient Estimation of Small P-values with the Cross-Entropy Method: Applications in Genomic Data Analysis</span><br class="auto-style3"><span class="auto-style2">Bioinformatics</span><span class="auto-style3">, 35 (14): 2441-2448. </span><span class="s3">[<a href="https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/bty1005/5232222"><span class="auto-style3">online</span></a>][<a href="https://arxiv.org/abs/1803.03373"><span class="auto-style3">arXiv</span></a><span class="auto-style3">][<a href="https://github.com/shilab2017/MCMC-CE-codes"><span class="auto-style3">code</span></a>]</span></span></p>
  </li>
  <li>
          <p class="p2">
              <span class="auto-style3">Liu K, Ye J, Yang Y, Shen L, Jiang H (2019) A Unified Model for Joint Normalization and Differential Gene Expression Detection in RNA-Seq data </span><br class="auto-style3"><span class="auto-style2">IEEE/ACM Transactions on Computational Biology and Bioinformatics</span><span class="auto-style3">, 16 (2): 442-454. </span><span class="s3">[<a href="http://ieeexplore.ieee.org/document/8249873/"><span class="auto-style3">online</span></a>][<a href="https://arxiv.org/abs/1610.04078"><span class="auto-style3">arXiv</span></a><span class="auto-style3">][<a href="ELMSeq/">code</a>]</span></span></p>
  </li>
  <li>
          <p class="p2">
              <span class="auto-style3">Liu T, Jiang H (2019) Minimizing Sum of Truncated Convex Functions and its Applications</span><br class="auto-style3"><span class="auto-style2">Journal of Computational and Graphical Statistics</span><span class="auto-style3">, 28 (1): 1-10. </span><span class="s3">[<a href="http://www.tandfonline.com/doi/full/10.1080/10618600.2017.1390471"><span class="auto-style3">online</span></a>][<a href="https://arxiv.org/abs/1608.00236"><span class="auto-style3">arXiv</span></a><span class="auto-style3">][<a href="stcf/">code</a>]</span></span></p>
  </li>
 <li>
  <p class="p2">
      <span class="auto-style3">Segal B, Elliott M, Braun T, Jiang H (2018)  P-splines with an l1 penalty for repeated measures</span><br class="auto-style3"><span class="auto-style2">Electronic Journal of Statistics</span><span class="auto-style3">, 12 (2): 3554-3600. </span><span class="s3">[<a href="https://projecteuclid.org/euclid.ejs/1540951342"><span class="auto-style3">online</span></a>][<a href="https://arxiv.org/abs/1707.08933"><span class="auto-style3">arXiv</span></a><span class="auto-style3">]</span>[<a href="https://github.com/bdsegal/code-for-psplinesl1-paper"><span class="auto-style3">code</span></a>]</span></p>
</li>
<li>
     <p class="p2">
         <span class="auto-style3">Segal B, Braun T, Elliott M, Jiang H (2018) Fast Approximation of Small p-values in Permutation Tests by Partitioning the Permutations</span><br class="auto-style3"><span class="auto-style2">Biometrics</span><span class="auto-style3">, 74 (1): 196-206. [<a href="https://onlinelibrary.wiley.com/doi/10.1111/biom.12731/full">online</a>][<a href="http://arxiv.org/abs/1605.03992"><span class="auto-style3">arXiv</span></a><span class="auto-style3">][<a href="https://github.com/bdsegal/fastPerm">software</a>][<a href="https://github.com/bdsegal/code-for-fastPerm-paper">code</a>]</span></span></p>
 </li>
 <li>
     <p class="p2">
         <span class="s2"><span class="auto-style3">He K, Zhou X, Jiang H, Wen X, Li Y (2018) False discovery control for penalized variable selections with high-dimensional covariates</span><br class="auto-style3"><span class="auto-style2">Statistical Applications in Genetics and Molecular Biology</span><span class="auto-style3">, 17(6). </span> </span><span class="s3">[<a href="https://www.degruyter.com/document/doi/10.1515/sagmb-2018-0038/html?lang=de"><span class="auto-style3">online</span></a><span class="auto-style3">]</span></span></p>
 </li>
 <li>
     <p class="p2">
         <span class="s2"><span class="auto-style3">Zhao L, Wu W, Feng D, Jiang H, Nguyen X (2018) Bayesian Analysis of RNA-Seq Data Using a Family of Negative Binomial Models</span><br class="auto-style3"><span class="auto-style2">Bayesian Analysis</span><span class="auto-style3">, 13 (2): 411-436. </span> </span><span class="s3">[<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8052637/"><span class="auto-style3">online</span></a><span class="auto-style3">]</span></span></p>
 </li>
 <li>
      <p class="p2">
          <span class="auto-style3">Jiang H, Zhan T (2017) Unit-Free and Robust Detection of Differential Expression from RNA-Seq Data </span><br class="auto-style3"><span class="auto-style2">Statistics in Biosciences</span><span class="auto-style3">, 9: 178-199. <span class="s3">[<a href="http://em.rdcu.be/wf/click?upn=KP7O1RED-2BlD0F9LDqGVeSHRcKICTqq-2BE6WeJUt1ftvk-3D_7hhHBA9vEFYlmqAWJJRvgAwB-2B0k7uyI4OlQpDUwxk-2FIoYw2reiKevnrHzb83Vzm6X7PoBTIu7D2hVRoeNk3yWBF6H-2FJ3waUR-2FwGzs3qzVtvGtTGr8mcLIoAj2CL-2B0cZhbi6JKvzaPCnDYnzz05HEb0rYTc8S1phjBym0E4QQK4ChmTTpBWV1rrhO6ppancNljp0iwm-2FhEuizdAVTUqAQrw-3D-3D">online</a>]</span></span><span class="s3">[<a href="http://arxiv.org/abs/1405.4538"><span class="auto-style3">arXiv</span></a><span class="auto-style3">][<a href="rseqrobust/">code</a>]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="auto-style3">Jiang H, Mu JC, Yang K, Du C, Lu L, Wong WH (2016) Computational Aspects of Optional Polya Tree</span><br class="auto-style3"><span class="auto-style2">Journal of Computational and Graphical Statistics</span><span class="auto-style3">, 25 (1): 301-320. </span><span class="s3">[<a href="http://www.tandfonline.com/doi/abs/10.1080/10618600.2014.1002927"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="http://arxiv.org/abs/1309.5489"><span class="auto-style3">arXiv</span></a><span class="auto-style3">][</span><a href="http://www.stanford.edu/group/wonglab/opt_comp/"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="auto-style3">Shi Y, Chinnaiyan AM, Jiang H (2015) rSeqNP: A non-parametric approach for detecting differential expression and splicing from RNA-Seq data</span><br class="auto-style3"><span class="auto-style2">Bioinformatics</span><span class="auto-style3">, 31 (13): 2222-2224</span><span class="auto-style2">. </span><span class="s3">[<a href="http://bioinformatics.oxfordjournals.org/content/31/13/2222"><span class="auto-style3">online</span></a><span class="auto-style3">][</span></span><a href="rseqnp/"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></p>
  </li>
          <li>
      <p class="p2">
          <span class="auto-style3">Jiang H, Salzman J (2015) A penalized likelihood approach for robust estimation of isoform expression</span><br class="auto-style3"><span class="auto-style2">Statistics and Its Interface</span><span class="auto-style3">, 8 (4): 437-445. </span> <span class="s3">[<a href="http://intlpress.com/site/pub/pages/journals/items/sii/content/vols/0008/0004/a003/index.html"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="http://arxiv.org/abs/1310.0379"><span class="auto-style3">arXiv</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Lu L, Jiang H, Wong WH (2013) Multivariate density estimation by Bayesian Sequential Partitioning</span><br class="auto-style3"><span class="auto-style2">Journal of the American Statistical Association</span><span class="auto-style3">, 108 (504): 1402-1410. </span> </span><span class="s3">[<a href="http://www.tandfonline.com/doi/full/10.1080/01621459.2013.813389"><span class="auto-style3">online</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
<p class="p2">
  <span class="s2"><span class="auto-style3">Shi Y, Jiang H (2013) rSeqDiff: Detecting differential isoform expression from RNA-Seq data using hierarchical likelihood ratio test</span><br class="auto-style3"><span class="auto-style2">PLoS One</span><span class="auto-style3">, 8 (11): e79448. </span> </span><span class="s3">[<a href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0079448"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="rseqdiff/"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></span></p>
</li>
<li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Jiang H, Salzman J (2012) Statistical properties of an early stopping rule for resampling-based multiple testing</span><br class="auto-style3"><span class="auto-style2">Biometrika</span><span class="auto-style3">, 99 (4): 973-980. </span> </span><span class="s3">[<a href="http://biomet.oxfordjournals.org/content/99/4/973"><span class="auto-style3">online</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Mu JC, Jiang H, Kiani A, Mohiyuddin M, Asadi NB, Wong WH (2012) Fast and Accurate Read Alignment for Resequencing</span><br class="auto-style3"><span class="auto-style2">Bioinformatics</span><span class="auto-style3">, 28 (18): 2366-2373. </span> </span><span class="s3">[<a href="https://academic.oup.com/bioinformatics/article-abstract/28/18/2366/253622"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="https://web.stanford.edu/group/wonglab/seqalto/"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Salzman J, Jiang H, Wong WH (2011) Statistical Modeling of RNA-Seq Data</span><br class="auto-style3"><span class="auto-style2">Statistical Science</span><span class="auto-style3">, 26 (1): 62-83. </span> </span><span class="s3">[<a href="http://projecteuclid.org/DPubS?verb=Display&amp;version=1.0&amp;service=UI&amp;handle=euclid.ss/1307626566&amp;page=record"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="http://arxiv.org/abs/1106.3211"><span class="auto-style3">arXiv</span></a><span class="auto-style3">][</span><a href="https://statistics.stanford.edu/sites/default/files/BIO%20252.pdf"><span class="auto-style3">technical report</span></a><span class="auto-style3">][</span><a href="rseq/"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Li J, Jiang H, Wong WH (2010) Modeling Non-Uniformity in Short-Read Rates in RNA-Seq Data</span><br class="auto-style3"><span class="auto-style2">Genome Biology</span><span class="auto-style3">, 11: R50.</span></span><span class="s3"><span class="auto-style3"> [</span><a href="http://genomebiology.com/2010/11/5/R50"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="http://cran.r-project.org/web/packages/mseq/index.html"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Au KF, Jiang H, Lin L, Xing Y, Wong WH (2010) Detection of Splice Junctions from Paired-End RNA-Seq Data by SpliceMap</span><br class="auto-style3"><span class="auto-style2">Nucleic Acids Research</span><span class="auto-style3">, 38 (14): 4570-4578. </span> </span><span class="s3">[<a href="http://nar.oxfordjournals.org/content/38/14/4570"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="http://www.stanford.edu/group/wonglab/SpliceMap/"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Jiang H, Wang F, Dyer NP, Wong WH (2010) CisGenome Browser: A flexible tool for genomic data visualization</span><br class="auto-style3"><span class="auto-style2">Bioinformatics</span><span class="auto-style3">, 26 (14): 1781–1782. </span> </span><span class="s3">[<a href="https://academic.oup.com/bioinformatics/article/26/14/1781/178901"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="browser/"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Jiang H, Wong WH (2009) Statistical Inferences for Isoform Expression in RNA-Seq</span><br class="auto-style3"><span class="auto-style2">Bioinformatics</span><span class="auto-style3">, 25 (8): 1026–1032. </span> </span><span class="s3">[<a href="http://bioinformatics.oxfordjournals.org/cgi/content/full/25/8/1026"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="download/rnaseq_preprint.pdf"><span class="auto-style3">preprint</span></a><span class="auto-style3">][</span><a href="rseq/"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Hiller D, Jiang H, Xu W, Wong WH (2009) Identifiability of Isoform Deconvolution from Junction Arrays and RNA-Seq</span><br class="auto-style3"><span class="auto-style2">Bioinformatics, </span><span class="auto-style3">25 (23): 3056-3059. </span> </span><span class="s3">[<a href="https://academic.oup.com/bioinformatics/article/25/23/3056/214567"><span class="auto-style3">online</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Kapur K, Jiang H, Xing Y, Wong WH (2008) Cross-Hybridization Modeling on Affymetrix Exon Arrays</span><br class="auto-style3"><span class="auto-style2">Bioinformatics, </span><span class="auto-style3">24 (24): 2887-2893. </span> </span><span class="s3">[<a href="https://academic.oup.com/bioinformatics/article/24/24/2887/197704"><span class="auto-style3">online</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Jiang H, Wong WH (2008) SeqMap: Mapping Massive Amount of Oligonucleotides to the Genome</span><br class="auto-style3"><span class="auto-style2">Bioinformatics, </span><span class="auto-style3">24 (20). </span> </span><span class="s3">[<a href="http://bioinformatics.oxfordjournals.org/cgi/content/short/24/20/2395"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="seqmap/"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Ji H, Jiang H, Ma W, Johnson D, Myers R, Wong WH (2008) An Integrated Software System for Analyzing ChIP-chip and ChIP-seq Data</span><br class="auto-style3"><span class="auto-style2">Nature Biotechnology,</span><span class="auto-style3"> 26: 1293-1300.</span></span><span class="s3"><span class="auto-style3"> [</span><a href="http://www.nature.com/nbt/journal/v26/n11/abs/nbt.1505.html"><span class="auto-style3">online</span></a><span class="auto-style3">][</span><a href="http://www.biostat.jhsph.edu/%7Ehji/cisgenome/"><span class="auto-style3">software</span></a><span class="auto-style3">]</span></span></p>
  </li>
  <li>
      <p class="p2">
          <span class="s2"><span class="auto-style3">Jiang H, Liu Y, Su Z (2008) Optimizing Chemotherapy Scheduling by Iteratively Solving a Recurrence Equation</span><br class="auto-style3"><span class="auto-style2">Algorithmic Operations Research,</span><span class="auto-style3"> 3(2).</span></span><span class="s3"><span class="auto-style3"> [</span><a href="https://www.erudit.org/en/journals/aor/2008-v3-n2-aor3_2/aor3_2art06/"><span class="auto-style3">online</span></a><span class="auto-style3">]</span></span></p>
  </li>
</ul>
</td>
</tr>
</tbody>
</table>
</body></html>
